R/zchunk_L222.en_transformation.R

```diff
@@ -81,248 +87,540 @@ module_energy_L222.en_transformation_DISABLED <- function(command, ...) {
+        filter(!is.na(L222.globaltech_retirement$half.life)) %>%
+        select(sector.name, subsector.name, technology, lifetime, steepness, half.life) ->
+        L222.GlobalTechSCurve_en
+    }
+
+    # L222.GlobalTechLifetime_en: Global tech lifetime
+    if (any(is.na(L222.globaltech_retirement$shutdown.rate) & is.na(L222.globaltech_retirement$half.life))){
+      L222.globaltech_retirement %>%
+        filter(is.na(L222.globaltech_retirement$shutdown.rate) & is.na(L222.globaltech_retirement$half.life)) %>%
+        select(sector.name, subsector.name, technology, lifetime) ->
+        L222.GlobalTechLifetime_en
+    }
+
+    # L222.GlobalTechProfitShutdown_en: Global tech profit shutdown decider and parameters
+    if (any(!is.na(L222.globaltech_retirement$median.shutdown.point))){
+      L222.globaltech_retirement %>%
+        filter(!is.na(L222.globaltech_retirement$median.shutdown.point)) %>%
+        select(sector.name, subsector.name, technology, median.shutdown.point, profit.shutdown.steepness) ->
+        L222.GlobalTechProfitShutdown_en
+    }
+
+    #2d. Calibration and region-specific data
+    #  generate base year calibrated outputs of gas processing by interpolating from historical values
+    L122.out_EJ_R_gasproc_F_Yh %>%
+      complete(nesting(GCAM_region_ID, sector, fuel), year = c(year, BASE_YEARS)) %>%
+      arrange(GCAM_region_ID, year) %>%
+      group_by(GCAM_region_ID, sector, fuel) %>%
+      mutate(value = approx_fun(as.numeric(year), value)) %>%
+      ungroup() %>%
+      filter(year %in% BASE_YEARS) %>%
+      # append region names
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      L222.out_EJ_R_gasproc_F_Yh_base
+
+    # append matching calibrated technology sector/subsector/technology to calibrated outputs of gas processing
+    calibrated_techs %>%
+      semi_join(L222.out_EJ_R_gasproc_F_Yh_base, by = c("sector", "fuel")) %>%
+      select(sector, fuel, supplysector, subsector, technology) %>%
+      left_join(L222.out_EJ_R_gasproc_F_Yh_base, by = c("sector", "fuel")) %>%
+      rename(stub.technology = technology) ->
+      L222.out_EJ_R_gasproc_F_Yh
+
+    # L222.StubTechProd_gasproc: calibrated output of gas processing technologies -- writes to all regions, adds calibrated tech info
+    A22.globaltech_coef %>%
+      filter(supplysector == "gas processing") %>%
+      write_to_all_regions(c("region", "supplysector", "subsector", "technology", "minicam.energy.input"), GCAM_region_names) %>%
+      rename(stub.technology = technology) %>%
+      repeat_add_columns(tibble("year" = BASE_YEARS)) %>%
+      left_join_error_no_match(L222.out_EJ_R_gasproc_F_Yh, by = c("region", "supplysector", "subsector", "stub.technology", "year")) %>%
+      # rounds outputs and adds year column for shareweights
+      mutate(calOutputValue = round(value, energy.DIGITS_CALOUTPUT), year.share.weight = year) %>%
+      select(-sector, -GCAM_region_ID, -fuel, -value, -minicam.energy.input) %>%
```

Comment:
In general please if possible `select` positively (inclusion instead of exclusion), as that's clearer.

Reply:
...sensibly. Dropping a column using `-x` is fine, obviously.